lib/inc/internal/util/agent.hpp

```diff
@@ -0,0 +1,22 @@
+/**
+ * @file
+ * Declares methods for Facter's integration with the puppet-agent package.
+ */
+#pragma once
+
+#include <leatherman/execution/execution.hpp>
+#include <string>
+
+namespace facter { namespace util { namespace agent {
```

Comment:
Perhaps just `facter::util` but call the function `agent_which`?  I don't really have strong opinion on this though; completely fine as is.